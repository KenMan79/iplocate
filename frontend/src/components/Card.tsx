import styled from "styled-components";

export const Card = styled.div`
  padding: 2rem 1rem;
  margin: 0;
  border-radius: 5px;
  background-color: var(--background-card);

  @media screen and (max-width: 450px) {
    padding: 1rem;
  }
`;

export const CardHeader = styled.h2`
  margin: 0;
  padding: 0;
  color: var(--text-primary);
  font-size: 1.8rem;
`;
