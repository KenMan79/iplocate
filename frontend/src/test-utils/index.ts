export * from "./test-utils";
export * from "./ipdata";
