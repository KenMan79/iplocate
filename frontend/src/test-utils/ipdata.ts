import { IP } from "types";

export const baseIP: IP = {
  continent: {
    code: "EU",
    geonameId: 6255148,
    names: {
      en: "Europe",
    },
  },
  country: {
    geonameId: 3017382,
    isInEuropeanUnion: true,
    isoCode: "FR",
    names: {
      en: "France",
    },
  },
  traits: {
    isAnonymous: false,
    isAnonymousProxy: false,
    isAnonymousVpn: false,
    isHostingProvider: false,
    isLegitimateProxy: false,
    isPublicProxy: false,
    isResidentialProxy: false,
    isSatelliteProvider: false,
    isTorExitNode: false,
    ipAddress: "2.2.2.2",
    network: "2.2.0.0/18",
  },
  city: {},
  location: {
    accuracyRadius: 500,
    latitude: 48.8582,
    longitude: 2.3387,
    timeZone: "Europe/Paris",
  },
};
