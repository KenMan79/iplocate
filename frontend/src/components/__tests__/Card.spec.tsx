import React from "react";
import { render } from "test-utils/test-utils";
import { Card } from "../Card";

it("matches snapshot", () => {
  const { asFragment } = render(<Card />);
  expect(asFragment()).toMatchSnapshot();
});
