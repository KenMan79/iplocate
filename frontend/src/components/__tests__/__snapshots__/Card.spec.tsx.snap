// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  padding: 2rem 1rem;
  margin: 0;
  border-radius: 5px;
  background-color: var(--background-card);
}

@media screen and (max-width:450px) {
  .c0 {
    padding: 1rem;
  }
}

<div
    class="c0"
  />
</DocumentFragment>
`;
